src/mds/MDSRank.cc

```diff
@@ -1771,9 +1774,6 @@ void MDSRank::starting_done()
   ceph_assert(is_starting());
   request_state(MDSMap::STATE_ACTIVE);
 
-  auto sle = mdcache->create_subtree_map();
-  mdlog->submit_entry(sle);
-
   // sync snaptable cache
   snapclient->sync(new C_MDSInternalNoop);
```

Comment:
While you're here, we should also require the snapclient is sync'd before entering active. Please also move this and use `MDSGatherBuilder` to trigger `::starting_done`.

Reply:
I leave it here because the order is the same as in creating_done. Should that be fixed as well or only starting state has to sync snap cache before requesting active?

Reply:
both should be fixed I think.

Reply:
It seems that snapclient sync cannot be called before requesting active. It doesn't work for rank that also runs as tableserver. snapclient->sync won't finish until MDSTableClient::server_ready is true, [here](https://github.com/ceph/ceph/blob/main/src/mds/SnapClient.cc#L154).
Yet server_ready is set only when [state of tabeserver's rank > STATE_RESOLVE](https://github.com/ceph/ceph/blob/main/src/mds/MDSRank.cc#L2458). Both STATE_STARTING and STATE_CREATING < STATE_RESOLVE. Therefore, the gather will block here.

I'm not familiar with snap client, but what snapclient->sync does seems to only query state from snap server which is only available after resolving state. It may have nothing to do with STARTING and CREATING state.

Reply:
You're right, let's not block this due to this.